scripts/prepare/tsc.ts

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env ../../node_modules/.bin/ts-node
+
+import { join } from 'path';
+import fs, { move } from 'fs-extra';
+import * as ts from 'typescript';
+import glob from 'glob';
+import { exec } from '../utils/exec';
+
+const hasFlag = (flags: string[], name: string) => !!flags.find((s) => s.startsWith(`--${name}`));
+
+const run = async ({ cwd, flags }: { cwd: string; flags: string[] }) => {
+  const {
+    bundler: { pre, post, tsConfig: tsconfigPath = 'tsconfig.json' },
+  } = await fs.readJson(join(cwd, 'package.json'));
+
+  if (pre) {
+    await exec(`node -r ${__dirname}/../node_modules/esbuild-register/register.js ${pre}`, { cwd });
+  }
+
+  const reset = hasFlag(flags, 'reset');
+  const watch = hasFlag(flags, 'watch');
+  // const optimized = hasFlag(flags, 'optimized');
```

Comment:
Typescript has no concept of minification, so we'd have to use another tool on top, which might break stuff.. so I'm going to say let's just skip it for now, not take the risk.